packages/components/src/Typeahead/__snapshots__/Typeahead.snapshot.test.js.snap

```diff
@@ -1,5 +1,167 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`Typeahead custom renders should render custom element 1`] = `
+<div
+  aria-expanded={true}
+  aria-haspopup="listbox"
+  aria-owns="react-autowhatever-my-search"
+  className="theme-tc-typeahead-container tc-typeahead-container theme-container-open tc-typeahead-container-open"
+  role="combobox"
+>
+  <div
+    className="theme-typeahead-input-icon tc-typeahead-typeahead-input-icon"
+  >
+    <label
+      className="control-label sr-only"
+      htmlFor="react-autowhatever-my-search-input"
+    >
+      Search
+    </label>
+    <input
+      aria-activedescendant={null}
+      aria-autocomplete="list"
+      aria-controls="react-autowhatever-my-search"
+      autoComplete="off"
+      autoFocus={false}
+      className="theme-typeahead-input tc-typeahead-typeahead-input form-control"
+      disabled={false}
+      id="my-search"
+      onBlur={[Function]}
+      onFocus={[Function]}
+      readOnly={false}
+      type="text"
+    />
+  </div>
+  <button>
+    Do some stuff
+  </button>
+</div>
+`;
+
+exports[`Typeahead custom renders should render custom element when loading 1`] = `
+<div
+  aria-expanded={false}
+  aria-haspopup="listbox"
+  aria-owns="react-autowhatever-my-search"
+  className="theme-tc-typeahead-container tc-typeahead-container theme-loading"
+  role="combobox"
+>
+  <div
+    className="theme-typeahead-input-icon tc-typeahead-typeahead-input-icon"
+  >
+    <label
+      className="control-label sr-only"
+      htmlFor="react-autowhatever-my-search-input"
+    >
+      Search
+    </label>
+    <input
+      aria-activedescendant={null}
+      aria-autocomplete="list"
+      aria-controls="react-autowhatever-my-search"
+      autoComplete="off"
+      autoFocus={false}
+      className="theme-typeahead-input tc-typeahead-typeahead-input form-control"
+      disabled={false}
+      id="my-search"
+      onBlur={[Function]}
+      onFocus={[Function]}
+      readOnly={false}
+      type="text"
+      value="he"
+    />
+  </div>
+  <div
+    className="theme-items-container tc-typeahead-items-container theme-items-container theme-container-open"
+    id="react-autowhatever-my-search"
+    role="listbox"
+  >
+    <div
+      className="theme-is-loading is-loading"
+    >
+      Loading...
```

Comment:
The snapshot doesn't match the test.
Instead of snapshoting, can you add an id or a classname to your custom renderers, and select the element, either snapshoting only those, or checking that they are present.
It's quite hard the see in the whole snapshots what you need to check.

In general, snapshots are good to check the whole structure. But when you test a specific case that will change a specific part, avoid snapshoting the whole component, try to pinpoint what you need.

Reply:
Hi @jsomsanith ,
I've replaced snapshot tests with pinpoint test.